### ISSUE TITLE
Fifo_1 Access denied?

### ISSUE THREAD

**Issue by bdmessenger** [hellslinger/hdpvr2_pipe#1]
This is what I got after I ran the script for the first time and for the next few times. I'm not sure what I'm doing wrong. Despite this message:   ./run_pipe.sh: line 55: fifo_1: Permission denied

It created fifo_0 & fifo_1 files when I ran the script for the first time. This was after I modified the script to set the directory for the hdpvr2 executable file. Also no capture preview on OBS.

Please help.

-Brant

**Comment by hellslinger** [comment]
Try: 
sudo chown `whoami` fifo_1
chmod 754 fifo_1

You won't be able to see the preview until bigpipe is running.

Let me know your results.

**Comment by bdmessenger** [comment]
I was able to enter in the commands you provided but it still gives me permission denied for fifo_1.

//--------------terminal_display---------------------//

here
./run_pipe.sh: line 55: fifo_1: Permission denied
000000.000000 ======== restarting in 5 sec =============
./run_pipe.sh: line 81: kill: (4553) - No such process

//-------------------------------------------------------//

I don't know if this helps but I'm running Ubuntu xenial on my chromebook. (Still new at learning Ubuntu)

**Comment by hellslinger** [comment]
I've never seen a user being unable to write to a FIFO he created, so I'm wondering if being on a Chromebook might be the reason. Can you post the output of ls -al in the directory where you're running everything from?

**Comment by bdmessenger** [comment]
Sorry for the long delay. On my Chromebook, I couldn't figure out the solution for that problem, but I ended up reinstalling another Ubuntu version and was able to enter the commands to bypass the restriction except it was for the run_pipe script instead. The script doesn't have the permission denied line being shown. However, towards the end of running the run_pipe script, it gives me this:

./run_pipe.sh: line 83: 31195 Terminated              $PIPE_PATH "$FIFO_IN" > "$FIFO_OUT"
here

I don't know if this is a good sign, but other than that no other errors have occurred.
When I ran the command line of "./bigpipe fifo_0 fifo_1" with OBS running in the background, the command line starts outputting dashes, one by one and fills the window. I have no preview of my capture card on OBS. I'm not sure if I'm following the procedures correctly.

**Comment by hellslinger** [comment]
Sounds like you're getting closer to have it working. I just got settled after a big move, so I'll have time this weekend to bring out my capture card and run things again.

A couple of questions in the meantime: 
What version of Ubuntu are you using? lsb_release -a
What version of kernel? uname -a
If you're using an ARM based system, I will need to test on my RPi.

**Comment by bdmessenger** [comment]
I'm using the Ubuntu 16.04.6 LTS version.
The kernel is x86_64.
I looked up my Chromebook more and I believe the codename for it is Kip.

**Comment by bdmessenger** [comment]
Which Linux were you using when you made the code?

**Comment by hellslinger** [comment]
This was all tested on Ubuntu 18.04. I will test again this week.

**Comment by hellslinger** [comment]
I was able to make everything work on a regular desktop PC, and I updated the readme. I believe that chromebooks may use SELinux which can prevent users from creating pipes by limiting certain systemcalls. 

Are you able to simply run ./HDPVR2_TestApp and get the device to capture video to a file? Be sure to use the right flag for HDMI and Composite/Component.

**Comment by bdmessenger** [comment]
Yes, I've been able to record footage from consoles using HDPVR2_TestApp to create video files. I ended up overwriting ChromeOS with Ubuntu since I'm able to boot to hard drive thanks to Seabios, but I couldn't get your method to work on Ubuntu 18.04.2. I gave up on this a while ago and needed to fix audio issues of my laptop so I ended switching over Linux Mint 19.1. Luckily, I did fix my audio on Mint. Other than that I haven't tried it on Mint but I can do so later on.
